### PR TITLE
add type and stub out add-product

### PR DESCRIPTION
**Pull request by mikeyhc**

### CODE REVIEW

File: add-product/main.c [hunk 1]
```diff
@@ -1,8 +1,25 @@
 #include <stdio.h>
+#include <stdlib.h>
```
Comment (kouKatsumi) [inline add-product/main.c:2]: What order we want to maintain in imports? (alphabet?)

File: add-product/main.c [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+main(int argc, char **argv)
 {
-	printf("hello, world\n");
+	unsigned price, qty, cost;
+	const char *type;
+	ARGBEGIN {
+	case 'h':
+		usage();
+		exit(0);
+		break;
```
Comment (kouKatsumi) [inline add-product/main.c:22]: Do we need a break after exit?

File: add-product/main.c [hunk 3]
```diff
@@ -1,8 +1,25 @@
 #include <stdio.h>
+#include <stdlib.h>
+
+#include "../common/arg.h"
+#include "../common/util.h"
+
+static void
+usage(void)
+{
+	printf("%s [-h] [-p PRICE] [-q QTY] [-c COST] -t TYPE NAME\n", argv0);
```
Comment (kouKatsumi) [inline add-product/main.c:10]: No -d for this one?

File: add-type/main.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	printf("usage: %s [-h] [-d DIR] TYPE\n", argv0);
+}
+
+int
+main(int argc, char **argv)
+{
+	char *path;
+	const char *type;
+
+	path = NULL;
```
Comment (kouKatsumi) [inline add-type/main.c]: Assign with declaration?

File: common/session.c [hunk 5]
```diff
@@ -52,7 +52,31 @@ create_session(sqlite3 *db)
...
+unsigned long
+current_sesion(sqlite3 *db)
+{
+	sqlite3_stmt *stmt;
+	const char *unused;
+	long ret;
+	int rc;
+
+	rc = sqlite3_prepare_v2(db,
+			"SELECT rowid FROM session WHERE end IS NULL",
```
Comment (kouKatsumi) [inline common/session.c:69]: Not convinced with indentation.

File: add-product/main.c (hunk 3 above)
Comment (mikeyhc) [reply add-product/main.c:10]: add product is incomplete, can be added

File: add-product/main.c (hunk 2 above)
Comment (mikeyhc) [reply add-product/main.c:22]: normally not, but I maintain it in case of refactoring later

File: add-type/main.c (hunk 4 above)
Comment (mikeyhc) [reply add-type/main.c]: prefer to align all imports then do declaration, happy to discuss though

File: common/session.c (hunk 5 above)
Comment (mikeyhc) [reply common/session.c:69]: standard C indentation

File: add-product/main.c (hunk 1 above)
Comment (mikeyhc) [reply add-product/main.c:2]: alphabetic libraries, then alphabetic local

File: add-product/main.c (hunk 2 above)
Comment (kouKatsumi) [reply add-product/main.c:22]: I think there won't be any other case for -h. Would just do exit.

File: add-type/main.c (hunk 4 above)
Comment (kouKatsumi) [reply add-type/main.c]: By standard: "Assign at declaration when possible"

**Review by kouKatsumi — APPROVED** [review]

**Pull request merged**